spec/factories/events.rb

```diff
@@ -123,6 +123,12 @@
       end
     end
 
+    trait :with_enough_free_places do
+      after(:build) do |event|
+        event.max_participants = event.application_letters.size
```

Comment:
Before this changes we could set the max_participants to infinity to ensure that there are enough free spaces. Now we could set a big number and hope that there aren't enough free places, but I think this is even uglier.... Maybe it would be the best if we set this value in the tests by hand?